### PR TITLE
CI Integration for master, dev, PRs and custom branches

### PR DESCRIPTION
**Pull request by fbeltrao**
This PR contains the first version of the CI.

* Runs full ci every time master or dev changes
* Runs quick version (build + unit test) for PRs
* Runs full ci on PRs with Label "fullci" (if Label is added `Rebuild` in Azure DevOps it to run the full version)
* Can be manually triggered by team members to run on a personal branch (`FullCI=true` to deploy and test on real device)
* Can be run only for device tests (option `RunTestsOnly = true`)
* Can be run to test a only specific tests (option `AdditionalIntegrationTestArguments  = --filter OnlyThisTest`)

**Code changes**
* yaml build pipeline
* UDP logging: resolving ip address by dns name
* Test cases: more logging to help troubleshooting
* Using test_runner.sh to retry tests that failed because the device is not responsive

**Changes after deployment refactoring**

* AzureDevOps module is not built as part of the CI. It is pre-built for arm32v7 as there is no official image. AMD64 uses the official microsoft/vsts-agent image. To keep both images in sync exposed same set of environment variables expected by official image (VSTS_ACCOUNT, VSTS_AGENT)
* Updated VSTS agent version to 2.142.1
* Change example.env regarding AzureDevOps image and RESET_PIN
* Fixed typo small typo in readme.md
* Added GetSyncHeader back to class LoraPhysicalPayload, as simulator uses it
* Added support for AMD based test device
* Cleared log clearing in integration tests
* Increased maximum run time of CI tests to 60 minutes

### CODE REVIEW

File: LoRaEngine/modules/LoRaSimulator/LoRaSimulator/SimulatedDevice.cs [hunk 1]
```diff
@@ -91,7 +91,7 @@ public byte[] GetUnconfirmedDataUpMessage()
             Array.Reverse(_payload);
             // 0 = uplink, 1 = downlink
             int direction = 0;
-            LoRaPayloadData standardData = new LoRaPayloadData(_mhbr, _devAddr, _FCtrl, _FCnt, _Fopts, _FPort, _payload, direction);
+            LoRaPayloadData standardData = new LoRaPayloadData((LoRaPayloadData.MType)_mhbr[0], _devAddr, _FCtrl, _FCnt, _Fopts, _FPort, _payload, direction);
```
Comment (Mandur) [inline LoRaEngine/modules/LoRaSimulator/LoRaSimulator/SimulatedDevice.cs]: We chose to move away from _xxx notation, we should probably refactor this part of the solution

Comment (fbeltrao) [reply LoRaEngine/modules/LoRaSimulator/LoRaSimulator/SimulatedDevice.cs]: I agree. My suggestion would be to integrate code analysis as warning. Then start fixing the warnings as we touch classes/methods.

What do you think?

Comment (ronniesa) [reply LoRaEngine/modules/LoRaSimulator/LoRaSimulator/SimulatedDevice.cs]: yes I did play around StyleCop roslyin analyzer... somtihng like 700 errors but we should start using it. Or do you have any other preference?

Comment (fbeltrao) [reply LoRaEngine/modules/LoRaSimulator/LoRaSimulator/SimulatedDevice.cs]: On top of my head I know FxCop and StyleCop analysers. If you agree, I can start working on it after the CI is done.

Comment (Mandur) [reply LoRaEngine/modules/LoRaSimulator/LoRaSimulator/SimulatedDevice.cs]: 🎉

**Comment by fbeltrao** [comment]
@ronniesa, @Mandur I have updated the repository with the required changes.
CI tests are working! 

I have done a forced push due to a lot commit noise. This PR is now ready for approval and merging

**Pull request merged**